src/renderer/src/hooks/useMount.ts

```diff
@@ -29,13 +29,21 @@ export const useMount = (
 }
 
 export const useMountUnsafe = (effect: EffectCallback) => {
-  const initialized = useRef(false)
+  const numberOfRender = useRef(0)
+  const unmountEffectRef = useRef<ReturnType<EffectCallback>>()
 
   useEffect(() => {
-    if (initialized.current) return
+    numberOfRender.current += 1
 
-    initialized.current = true
-    return effect()
+    if (numberOfRender.current <= 1) {
+      unmountEffectRef.current = effect()
+    }
+
+    return () => {
+      if (numberOfRender.current < 1 && unmountEffectRef.current) {
+        unmountEffectRef.current()
+      }
+    }
```

Comment:
Why have you changed this hook?

Reply:
I left some comments in the code to explain why it needs to be like this